esp-hal/src/dma/mod.rs

```diff
@@ -397,6 +399,24 @@ macro_rules! dma_circular_descriptors {
     };
 }
 
+/// Allocates a DMA buffer with a specific size, aligned to 4 bytes
+#[doc(hidden)]
+#[macro_export]
+macro_rules! allocate_dma_buffer {
```

Comment:
not sure I like this name as it implies it's using the allocator... maybe `declare_dma_buffer`?

Reply:
Works for me